src/node_binding-inl.h

```diff
@@ -0,0 +1,57 @@
+#ifndef SRC_NODE_BINDING_INL_H_
+#define SRC_NODE_BINDING_INL_H_
+
```

Comment:
`#include "node_binding.h"` here - the rule is that you can include the -inl.h variant without having to include the main header file as well.

Then, include this file only in the compilation units that actually use it.

That said, I recommend putting this in a .cc file instead of a header file because there's really no need to inline it in every compilation unit where it's used, that only leads to code bloat.

Reply:
TBH I'd rather keep it `inline`. `Open()` is only called from `node_binding.cc` and `Close()` is only called from `node_binding.cc` and `env.cc` and, if we assume a future where many worker threads come and go the indirection of at least two function calls per thread per native addon may accumulate.

Reply:
That's not what `inline` does. It's not a compiler hint, it's a linkage directive; it means the function or method won't be visible outside the compilation unit.

That aside, two function calls aren't going to show up in profiles if you're also doing things like manipulating linked lists.

Reply:
@bnoordhuis I've moved the definitions of the functions back into `src/node_binding.cc`.